core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -34,8 +29,7 @@
 /**
  * A dynamic {@link EndpointGroup}. The list of {@link Endpoint}s can be updated dynamically.
  */
-public class DynamicEndpointGroup implements EndpointGroup {
-    private final Set<Consumer<Collection<Endpoint>>> updateListeners = new CopyOnWriteArraySet<>();
+public class DynamicEndpointGroup extends AbstractListenable<Collection<Endpoint>> implements EndpointGroup {
```

Comment:
Because I would like to reuse result of `List<Endpoint> endpoints`, I prefer use List or Collection. What do you think?